linera-core/src/client/mod.rs

```diff
@@ -487,13 +476,10 @@ pub struct ChainClientOptions {
 /// * As a rule, operations are considered successful (and communication may stop) when
 ///   they succeeded in gathering a quorum of responses.
 #[derive(Debug)]
-pub struct ChainClient<ValidatorNodeProvider, Storage>
-where
-    Storage: linera_storage::Storage,
-{
+pub struct ChainClient<Env: Environment> {
```

Comment:
To be clear, we still would like to remove the trait constraint (forced by async graphql IIRC) eventually.

Reply:
nit: Could we use `E` instead of `Env` ?

Reply:
Right, absolutely.  There's a neat trick for that: wherever we have

```rust
struct Foo<T: HasBar> {
    x: T,
    y: T::Bar,
}
```

we can transform it into:

```rust
struct Foo<T, TBar = <T as HasBar>::Bar> {
  x: T,
  y: TBar,
}
```

with no loss of ergonomics for the user.  But currently it's not worth doing this because the bounds are forced anyway by both `async-graphql` and also `linera-views`, which requires a type that `: Context` in its views, propagating the bound up from the `chain_worker`.

Reply:
On `E` rather than `Env`: we use (and Rust uses, generally) `E` to mean `Error` a lot, so I wanted to distinguish this parameter.